contracts/DebtToken.sol

```diff
@@ -120,43 +131,88 @@ contract DebtToken is MintableNonFungibleToken, Pausable {
      */
-    function _clearApprovalAndTransfer(
-        address _from,
-        address _to,
-        uint _tokenId
-    )
-        internal
+    function approve(address _to, uint _tokenId)
+        public
         whenNotPaused
     {
-        super._clearApprovalAndTransfer(_from, _to, _tokenId);
+        super.approve(_to, _tokenId);
     }
 
     /**
-     * We override the core approvals method of the parent non-fungible token
+     * We override setApprovalForAll method of the parent ERC721Token
      * contract to allow its functionality to be frozen in the case of an emergency
      */
-    function _approve(address _to, uint _tokenId)
-        internal
+    function setApprovalForAll(address _to, bool _approved)
+        public
         whenNotPaused
     {
-        super._approve(_to, _tokenId);
+        super.setApprovalForAll(_to, _approved);
     }
 
+    /**
+     * Support deprecated ERC721 method
+     */
+    function transfer(address _to, uint _tokenId)
+        public
+    {
+        safeTransferFrom(msg.sender, _to, _tokenId);
+    }
 
     /**
-     * We oveerride the core ownership transfer method of the parent non-fungible token
-     * contract so that it mutates the debt registry every time a token is transferred
+     * We override transferFrom methods of the parent ERC721Token
+     * contract to allow its functionality to be frozen in the case of an emergency
      */
-    function _setTokenOwner(uint _tokenId, address _to)
+    function transferFrom(address _from, address _to, uint _tokenId)
+        public
+        whenNotPaused
+    {
+        _modifyBeneficiary(_tokenId, _to);
+        super.transferFrom(_from, _to, _tokenId);
+    }
+
+    /**
+     * We override safeTransferFrom methods of the parent ERC721Token
+     * contract to allow its functionality to be frozen in the case of an emergency
+     */
+    function safeTransferFrom(address _from, address _to, uint _tokenId)
+        public
+        whenNotPaused
+    {
+        _modifyBeneficiary(_tokenId, _to);
+        super.safeTransferFrom(_from, _to, _tokenId);
+    }
+
+    /**
+     * We override safeTransferFrom methods of the parent ERC721Token
+     * contract to allow its functionality to be frozen in the case of an emergency
+     */
+    function safeTransferFrom(address _from, address _to, uint _tokenId, bytes _data)
+        public
+        whenNotPaused
+    {
+        _modifyBeneficiary(_tokenId, _to);
+        super.safeTransferFrom(_from, _to, _tokenId, _data);
+    }
+
+    function _mint(address _owner, uint _tokenId)
         internal
+    {
+        _modifyBeneficiary(_tokenId, _owner);
+        super._mint(_owner, _tokenId);
+    }
+
+    /**
+     * _modifyBeneficiary mutates the debt registry. This function should be
+     * called every time a token is transferred or minted
+     */
+    function _modifyBeneficiary(uint _tokenId, address _to)
+        private
```

Comment:
Careful -- this function is `private`, but is still externally callable.  Please add the `internal` keyword, so that it's not exposed to the public.